app/connectors/NpsConnector.scala

```diff
@@ -44,11 +45,13 @@ class NpsConnector @Inject()(http: HttpClient, applicationConfig: ApplicationCon
     implicit val desHeaderCarrier: HeaderCarrier = header(hc)
 
     http.GET[HttpResponse](url("/individuals/annual-tax-summary/" + ninoWithoutSuffix + "/" + TAX_YEAR)) recover {
-      case _: BadRequestException => HttpResponse(BAD_REQUEST)
-      case _: NotFoundException   => HttpResponse(NOT_FOUND)
+      case _: BadRequestException => HttpResponse(BAD_REQUEST, s"Bad request response in connector for $NINO")
+      case _: NotFoundException   => HttpResponse(NOT_FOUND, s"Not found response in connector for $NINO")
+      case _: UpstreamErrorResponse =>
```

Comment:
UpstreamErrorResponse will handle 4xx and 5xx, is that what you want here?

Reply:
Yes. If it is not caught in the first two cases it will either be 500 or 503. Not sure if you will remember me showing you this a while back, if I use anything other than UpstreamErrorResponse it throws an exception for a strange reason. So this UpstreamErrorResponse will catch all the 500's and 503's and return an InternalServerError to the service as we want without throwing any exceptions.

Reply:
So just so you are aware (feel free to write a test on this as well), if the response back is a 400 we get a BadRequestException, if its a 404 we get a NotFoundException. 
If it is any other 4xx response, we get a Upstream4xxResponse exception (which is a UpstreamErrorResponse sub class). If its a 5xx response, we get a Upstream5xxResponse exception (same sub class). 

My question is, are you sure you want to group 4xx and 5xx exceptions together?